### PR TITLE
gargoyle: init at 2017-03-12

### PR DESCRIPTION
**Pull request by orivej**
###### Motivation for this change

Gargoyle is a multi-format interactive fiction graphical interpreter. When launched without arguments it opens a file chooser dialog, you could try [Kerkerkruip](http://downloads.kerkerkruip.org/kerkerkruip-latest.zip) (after unpacking the zip) or [zork1](http://www.batmantis.com/zorks/zork1.z5). The last release is from 2011 but development is [ongoing](https://github.com/garglk/garglk/commits/master) and I'd prefer to package the latest code.

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/jam/default.nix [hunk 1]
```diff
@@ -1,11 +1,5 @@
...
 
-let
-  bindir = if stdenv.system == "i686-linux" then "bin.linuxx86"
-    else if stdenv.system == "x86_64-linux" then "bin.linux"
-    else throw "Unsupported platform by now";
-in
-
 stdenv.mkDerivation {
   name = "jam-2.5";
   src = fetchurl {
```
Comment (ndowens) [inline pkgs/development/tools/build-managers/jam/default.nix]: url = ftp://ftp.perforce.com/jam/${name}.tar;
Makes it easier for when version changes; change one version area verses two

File: pkgs/development/tools/build-managers/jam/default.nix [hunk 2]
```diff
@@ -17,13 +11,13 @@ stdenv.mkDerivation {
 
   installPhase = ''
     mkdir -p $out/bin
-    cp ${bindir}/jam $out/bin
+    cp bin.*/jam $out/bin
   '';
 
   meta = {
```
Comment (ndowens) [inline pkgs/development/tools/build-managers/jam/default.nix]: meta = with stdenv.lib; { 
     homepage = http://public.perforce.com/wiki/Jam;
￼      license = licenses.free;
￼      description = "Just Another Make";
￼     platforms = platforms.unix;
};
}

**Review by ndowens — CHANGES_REQUESTED** [review]

File: pkgs/development/tools/build-managers/jam/default.nix (hunk 2 above)
Comment (orivej) [inline pkgs/development/tools/build-managers/jam/default.nix]: Thanks, here I touched jam only to support darwin, I'm going to update jam in a separate review.

**Comment by orivej** [comment]
`jam` with darwin support has been merged, I updated this pull request.

File: pkgs/games/gargoyle/default.nix [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    export _LIBDIR=libexec/gargoyle
+    export GARGLKINI="$out/etc/garglk.ini"
+  '';
+
+  buildPhase = "jam -j$NIX_BUILD_CORES";
+
+  installPhase = if stdenv.isDarwin then (builtins.readFile ./darwin.sh) else ''
+    jam -j$NIX_BUILD_CORES install
+    mkdir -p "$out/bin"
+    ln -s ../libexec/gargoyle/gargoyle "$out/bin"
```
Comment (Mic92) [inline pkgs/games/gargoyle/default.nix]: the link target is not installed on linux:

```
$ [nix-shell:/nix/store/c2x2l9cy59qr20msbfjxaiji90906a8x-gargoyle-2017-03-12]$ ls -la bin/gargoyle
lrwxrwxrwx 1 root root 28 Jan  1  1970 bin/gargoyle -> ../libexec/gargoyle/gargoyle
$ bin/gargoyle                                                                      
bash: bin/gargoyle: No such file or directory
```

Comment (orivej) [inline pkgs/games/gargoyle/default.nix]: Oops, lost it during refactoring. Fixed by moving environment variable setting from `preBuild` back to `buildPhase` and `installPhase`.

Comment (Mic92) [inline pkgs/games/gargoyle/default.nix]: problem still persists:

```
$ find .
./etc
./etc/garglk.ini
./bin
./bin/gargoyle
./share
./share/icons
./share/icons/hicolor
./share/icons/hicolor/32x32
./share/icons/hicolor/32x32/apps
./share/icons/hicolor/32x32/apps/gargoyle-house.png
./share/applications
./share/applications/gargoyle.desktop
$ ls -la bin/gargoyle                                        
lrwxrwxrwx 1 root root 28 Jan  1  1970 bin/gargoyle -> ../libexec/gargoyle/gargoyle
```

Comment (orivej) [inline pkgs/games/gargoyle/default.nix]: Are you sure that you built the current revision? Last Travis check confirms that libexec is there: https://travis-ci.org/NixOS/nixpkgs/jobs/212810559#L3500

Comment (Mic92) [inline pkgs/games/gargoyle/default.nix]: no, you are right. I updated my pull request branch not correctly.

**Comment by Mic92** [comment]
Thanks!

**Pull request merged**